### PR TITLE
fix: only check timestamped forks after the merge

### PR DESCRIPTION
**Pull request by TovarishFin**
The current implementation does not take into account that timestamp forks should ALWAYS be applied after block number forks. This fixes that.

reference: https://eips.ethereum.org/EIPS/eip-6122

### CODE REVIEW

File: crates/primitives/src/chain/spec.rs [hunk 1]
```diff
@@ -224,29 +224,39 @@ impl ChainSpec {
...
-            };
-
-            if cond.active_at_head(head) {
-                if value != current_applied_value {
-                    curr_forkhash += value;
-                    current_applied_value = value;
+            if let ForkCondition::Block(block) = cond {
+                if cond.active_at_head(head) && block != current_applied {
+                    forkhash += block;
+                    println!("block: {} forkhash: {:#?}", block, forkhash);
```
Comment (onbjerg) [inline crates/primitives/src/chain/spec.rs:237]:
```suggestion

```

File: crates/primitives/src/chain/spec.rs [hunk 2]
```diff
@@ -224,29 +224,39 @@ impl ChainSpec {
...
             }
         }
-        ForkId { hash: curr_forkhash, next: 0 }
+
+        // timestamp are ALWAYS applied after the merge.
+        for (_, cond) in self.forks_iter() {
+            if let ForkCondition::Timestamp(timestamp) = cond {
+                if cond.active_at_head(head) && timestamp != current_applied {
+                    forkhash += timestamp;
+                    println!("block: {} forkhash: {:#?}", timestamp, forkhash);
```
Comment (onbjerg) [inline crates/primitives/src/chain/spec.rs:248]:
```suggestion

```

**Review by onbjerg — CHANGES_REQUESTED** [review]
Please add a test demonstrating that this is not the case currently and that this fixes it.

**Review by mattsse — COMMENTED** [review]
this looks right to me, however, this does not pass an existing forkid test.

pending @Rjected

**Comment by onbjerg** [comment]
Ping @Rjected, wdyt?

**Review by Rjected — APPROVED** [review]
Now this is good and passes tests, it originally missed the sepolia netsplit block edge case

**Review by mattsse — APPROVED** [review]
cool!
glad we can get this included

**Review by onbjerg — APPROVED** [review]

**Comment by gakonst** [comment]
Did we try this live? Want to be very sensitive to anything that touches this logic given it's in live p2p mode.

**Comment by Rjected** [comment]
> Did we try this live? Want to be very sensitive to anything that touches this logic given it's in live p2p mode.

Just tested this with sepolia, without any issues

**Pull request merged**